lib/shopify_app/controller_concerns/embedded_app.rb

```diff
@@ -0,0 +1,19 @@
+module ShopifyApp
+  module EmbeddedApp
+    extend ActiveSupport::Concern
+
+    included do
+      if ShopifyApp.configuration.embedded_app?
+        after_action :set_esdk_headers
+        layout 'embedded_app'
+      end
+    end
+
+    private
+
+    def set_esdk_headers
+      response.set_header('P3P', 'CP="Not used"')
+      response.default_headers.delete('X-Frame-Options')
```

Comment:
Eh, I thought about this a bit more and this should be safe to do since `X-Frame-Options` doesn't have anything to do with the server.